xr-active-switch.ts

```diff
@@ -0,0 +1,115 @@
+    @property.enum(['Activate', 'Deactivate', 'Toggle', 'Keep'])
+    ifInXR: string | number = 0;
+
+    /** Action to perform when XR session ends */
+    @property.enum(['Activate', 'Deactivate', 'Toggle', 'Keep'])
+    ifNotInXR: string | number = 1;
+
+    /** Scope of elements to affect */
+    @property.enum(['This Object', 'Children Only', 'This Object & Children'])
+    scope: string | number = 2;
+
+    private components: Component[] = [];
+    private onSessionStart!: () => void;
+    private onSessionEnd!: () => void;
+
+    start() {
+        this.components = [];
+        this.collectComponents();
+
+        // Initial state setup
+        if (this.engine.xr?.session) {
+            this.applyAction(this.ifInXR);
+        } else {
+            this.applyAction(this.ifNotInXR);
+        }
+
+        // Bind event handlers
+        this.onSessionStart = this.handleSessionStart.bind(this);
+        this.onSessionEnd = this.handleSessionEnd.bind(this);
+    }
+
+    onActivate() {
+        this.engine.onXRSessionStart.add(this.onSessionStart);
+        this.engine.onXRSessionEnd.add(this.onSessionEnd);
+    }
+
+    onDeactivate() {
+        this.engine.onXRSessionStart.remove(this.onSessionStart);
+        this.engine.onXRSessionEnd.remove(this.onSessionEnd);
+    }
+
+    private collectComponents() {
+        const mode = this.resolveEnumValue(this.scope, [
+            'This Object',
+            'Children Only',
+            'This Object & Children',
+        ]);
+
+        // Handle current object
+        if (mode === 0 || mode === 2) {
+            const comps = this.object
+                .getComponents()
+                .filter((c) => c.type !== XrActiveSwitch.TypeName);
+            this.components.push(...comps);
+        }
+
+        // Handle children
+        if (mode === 1 || mode === 2) {
+            this.processChildren(this.object);
+        }
+    }
+
+    private processChildren(obj: Object3D) {
+        for (const child of obj.children) {
+            const comps = child
+                .getComponents()
+                .filter((c) => c.type !== XrActiveSwitch.TypeName);
+            this.components.push(...comps);
+            this.processChildren(child); // Recurse through all descendants
+        }
+    }
+
+    private applyAction(action: string | number) {
+        const actionMap = ['Activate', 'Deactivate', 'Toggle', 'Keep'];
+        const actionIndex = this.resolveEnumValue(action, actionMap);
+
+        for (const comp of this.components) {
+            switch (actionIndex) {
+                case 0: // Activate
+                    comp.active = true;
+                    break;
+                case 1: // Deactivate
+                    comp.active = false;
+                    break;
+                case 2: // Toggle
+                    comp.active = !comp.active;
+                    break;
+                case 3: // Keep
+                default:
+                    break;
```

Comment:
```suggestion
            if(actionIndex == 0) {
                    comp.active = true;
            } else if(actionIndex == 1) {
                    comp.active = false;
            } else if(actionIndex == 2 {
                    comp.active = !comp.active;
            }
```

It's much shorter and a bit easier to read than a switch. If we'd expect more values to come at some point, the switch would definitely make sense, but in this case that's all there are going to be.

You may want to consider adding constants or an enum for the 0, 1, 2, (3) here, since it would make the code easier to read.